client/my-sites/plans/ecommerce-trial/upgrade-confirmation/index.tsx

```diff
@@ -15,8 +17,11 @@ import './style.scss';
 const TrialUpgradeConfirmation = () => {
 	const selectedSite = useSelector( getSelectedSite );
 	const translate = useTranslate();
-
-	const tasks = getConfirmationTasks( { translate } );
+	const siteId = selectedSite?.ID;
+	const hasWCPay = useSelector(
+		( state ) => siteId && isPluginActive( state, siteId, 'woocommerce-payments' )
+	) as boolean;
+	const tasks = getConfirmationTasks( { translate, hasWCPay } );
```

Comment:
Non-blocking thought: this might be worth wrapping in `useMemo()` so it only gets recomputed when `translate` or `hasWCPay` change.
```suggestion
	const tasks = useMemo( () => getConfirmationTasks( { translate, hasWCPay } ), [ hasWCPay, translate ] );
```